doc/arch/adr-007-pa-dep-client-app.md

```diff
@@ -0,0 +1,122 @@
+and would be much better be done in an automated fashion,
+where when users download the spreadsheet it is pre-populated with the right values in the right places.
+
+Since this requires having two effective areas of interest at once,
+this workflow does not fit neatly in MMW,
+which always expects a single area of interest.
+Rather than change MMW to accomodate this unusual and rare use case,
+it would be much simpler to create a new client app that is focused on this workflow.
+It will use the MMW API for all calculations.
+
+Given that we'll have a new client app,
+this is an opportunity to introduce a more modern tech stack on the front-end.
+This was recently done for [a similar project][beekeepers],
+where a new front-end stack was added within a Django app to sandbox it from the rest of the project.
+This document aims to discuss the pros and cons of different arrangements for the new client app,
+and pick one that satisfies all project constraints.
+
+The one major directive from the clients is that,
+from a user's perspective,
+the site should be deeply integrated with MMW.
+That is to say,
+it should use the same colors and branding,
+and not feel like its own project.
+
+From the development perspective,
+we should aim for using the latest front-end stacks,
+clean communication with the back-end API,
+and having a fast deployment pipeline.
+
+### Repository: Same vs Different
+
+Advantages of having the **same repository**:
+
+  * Familiarity, as it was done in the [Beekeepers][beekeepers] project
+  * All issues can be in one repo, simplifying project management
+  * Tasks that affect both front-end and back-end can be tracked in one commit or PR
+  * Simpler deployment, as front-end and back-end will always be in sync
+  * Styles can be easily reused between the two apps
+  * Easier to simulate production environments in development,
+    as we only need to start a single development environment.
+
+Advantages of having **different repositories**:
+
+  * Cleaner logical separation in the codebase
+  * Fewer cohabiting developer entry points, i.e. `package.json`, or multiple `node_modules`
+  * Does not require changing the `node` or `npm` version in MMW
+  * Does not require adding a Docker environment in MMW for building the front-end
+  * Faster deployment, as the client app can be deployed independently of the main app
+
+### Deployment: Paired vs Independent
+
+Advantages of **paired deployment**:
+
+  * Front-end and back-end will always be in sync
+  * Does not require extra configuration of AWS
+  * Does not need extra steps during deployment
+  * Easier to write end-to-end tests
+
+Advantages of **independent deployment**:
+
+  * Front-end can be iterated much faster, without the heavy deployment cycle of MMW
+
+### Access: Subdomain vs Subdirectory
+
+Advantages of **subdomain access**:
+
+  * More easily supports independent deployments
+
+Advantages of **subdirectory access**:
+
+  * Maintains MMW brand identity
```

Comment:
I will expand this section, but essentially, this is in response to the client constraint mentioned in the beginning:

> The one major directive from the clients is that, from a user's perspective, the site should be deeply integrated with MMW. That is to say, it should use the same colors and branding, and not feel like its own project.

Within MMW, we already have `/draw`, `/analyze`, `/monitor`, and `/model` routes. By keeping this app within something like `/bmp/`, we make it seem more a regular part of the app rather than something different.

Perhaps, given that the client constraint actively discourages subdomains, I should take them out of this document entirely.